# ðŸ”¬The test datasets of TPMA

### mt1x.tar.gz
 - Data source: [mt1x](http://lab.malab.cn/soft/halign/data/mt1x.zip)
 - Reference: Tanaka M., et al. (2004) Mitochondrial genome variation in eastern Asia and the peopling of Japan. Genome Res,14(10a), 1832-1850.
 - raw_data: 30 sequences were randomly selected from mt1x dataset, 4 times
 - msa_results: the alignments were generated using ClustalW2, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - reformalign_results: the alignments were optimized by ReformAlign.
 - Download: [mt1x.tar.gz](http://lab.malab.cn/soft/TPMA/data/mt1x.tar.gz)

### hvs_ii.tar.gz
 - Data source: [hvs_ii.fasta](http://lab.malab.cn/soft/TPMA/data/hvs_ii.fasta)
 - Reference: Ref: Ancient DNA Reveals Key Stages in the Formation of Central European Mitochondrial Genetic Diversity. DOI: 10.1126/science.1241844
 - raw_data: the dataset is divided into 10 sub-datasets, each sub-dataset has 10 sequences.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - Download: [16s_rRNA.tar.gz](http://lab.malab.cn/soft/TPMA/data/hvs_ii.tar.gz)

### 16s_rRNA.tar.gz
 - Data source: [16s_rRNA_small_test](http://lab.malab.cn/soft/halign/data/16ssmall.zip)
 - Reference: DeSantis, T. Z., et al.(2006) NAST: a multiple sequence alignment server for comparative analysis of 16S rRNA genes. Nucleic Acids Res, 34, W394-399.
 - raw_data: 100 sequences were randomly selected from 16s rRNA small dataset, 8 times.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - reformalign_results: the alignments were optimized by ReformAlign.
 - Download: [16s_rRNA.tar.gz](http://lab.malab.cn/soft/TPMA/data/16s_rRNA.tar.gz)

### 23s_rRNA.tar.gz
 - Data source: [23s_rRNA.tar.xz](http://lab.malab.cn/%7Etfr/HAlign3_testdata/23s_rRNA.tar.xz)
 - Reference: Furong Tang, Jiannan Chao, Yanming Wei, Fenglong Yang, Yixiao Zhai, Lei Xu, Quan Zou, HAlign 3: Fast Multiple Alignment of Ultra-Large Numbers of Similar DNA/RNA Sequences, Molecular Biology and Evolution, Volume 39, Issue 8, August 2022, msac166, https://doi.org/10.1093/molbev/msac166
 - raw_data: the dataset is divided into 10 sub-datasets, each sub-dataset has about 64 sequences
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - reformalign_results: the alignments were optimized by ReformAlign.
 - Download: [23s_rRNA.tar.gz](http://lab.malab.cn/soft/TPMA/data/23s_rRNA.tar.gz)

### SARS-CoV-2_20200301.tar.gz
 - Data source: [SARS-CoV-2_20200301.zip](http://lab.malab.cn/~zsl/Periodical_results/data/2019nCoVR_20200301.zip)
 - Reference: Yanming Wei, Quan Zou, Furong Tang, Liang Yu, WMSA: a novel method for multiple sequence alignment of DNA sequences, Bioinformatics, Volume 38, Issue 22, 15 November 2022, Pages 5019â€“5025, https://doi.org/10.1093/bioinformatics/btac658
 - raw_data: The dataset is divided into 4 sub-datasets, each sub-dataset has 39 sequences
 - msa_results: the alignments were generated using ClustalW2, Kalign3, MAFFT, MUSCLE3 and PCMA.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - Download: [SARS-CoV-2_20200301.tar.gz](http://lab.malab.cn/soft/TPMA/data/SARS-CoV-2_20200301.tar.gz)

### SARS-CoV-2_20200417.tar.gz
 - Data source: [SARS-CoV-2_20200417.7z](http://lab.malab.cn/~cjt/MSA/data/SARS-CoV-2_20200417.7z)
 - Reference: Yanming Wei, Quan Zou, Furong Tang, Liang Yu, WMSA: a novel method for multiple sequence alignment of DNA sequences, Bioinformatics, Volume 38, Issue 22, 15 November 2022, Pages 5019â€“5025, https://doi.org/10.1093/bioinformatics/btac658
 - raw_data: 100 sequences were randomly selected from SARS_CoV_20200417 dataset, 4 times
 - msa_results: the alignments were generated using ClustalW2, Kalign3, MAFFT, MUSCLE3 and PCMA.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - Download: [SARS-CoV-2_20200417.tar.gz](http://lab.malab.cn/soft/TPMA/data/SARS-CoV-2_20200417.tar.gz)

### 16s_simu.tar.gz
 - Data source: A sets of 16S rRNA genomes like was simulated using INDELible v1.03 (Fletcher and Yang 2009)
 - raw_data: same with the data source.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, HAlign3, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - reformalign_results: the alignments were optimized by ReformAlign.
 - Download: [16s_simu.tar.gz](http://lab.malab.cn/soft/TPMA/data/16s_simu.tar.gz)

### 23s_simu.tar.gz
 - Data source: A sets of Mycobacterium 23S rRNA like was simulated using INDELible v1.03 (Fletcher and Yang 2009)
 - raw_data: same with the data source.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, HAlign3, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - reformalign_results: the alignments were optimized by ReformAlign.
 - Download: [23s_simu.tar.gz](http://lab.malab.cn/soft/TPMA/data/23s_simu.tar.gz)

### CIPRES-128.tar.gz
 - Data source: [CIPRES-128](http://lab.malab.cn/soft/WMSA2/data/RNA-255.zip)
 - Reference: Chen J, Chao J, Liu H, Yang F, Zou Q, Tang F. WMSA 2: a multiple DNA/RNA sequence alignment tool implemented with accurate progressive mode and a fast win-win mode combining the center star and progressive strategies. Brief Bioinform. 2023 May 17:bbad190. doi: 10.1093/bib/bbad190. Epub ahead of print. PMID: 37200156.
 - raw_data: same with the data source.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, HAlign3, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - Download: [CIPRES-128.tar.gz](http://lab.malab.cn/soft/TPMA/data/CIPRES-128.tar.gz)

### CIPRES-256.tar.gz
 - Data source: [CIPRES-256](http://lab.malab.cn/soft/WMSA2/data/RNA-511.zip)
  - Reference: Chen J, Chao J, Liu H, Yang F, Zou Q, Tang F. WMSA 2: a multiple DNA/RNA sequence alignment tool implemented with accurate progressive mode and a fast win-win mode combining the center star and progressive strategies. Brief Bioinform. 2023 May 17:bbad190. doi: 10.1093/bib/bbad190. Epub ahead of print. PMID: 37200156.
 - raw_data: same with the data source.
 - msa_results: the alignments were generated using ClustalW2, Dialign-TX, Kalign3, MAFFT, MUSCLE3, MUSCLE5, PCMA, POA, T-Coffee and WMSA2.
 - tpma_results: the alignments obtained by TPMA.
 - mcoffee_results: the alignments obtained by M-Coffee.
 - Download: [CIPRES-256.tar.gz](http://lab.malab.cn/soft/TPMA/data/CIPRES-256.tar.gz)
